spec.html

```diff
@@ -326,29 +337,35 @@ <h1>%SegmentIteratorPrototype%.next( )</h1>
 
       <emu-clause id="sec-segment-iterator-prototype-following">
         <h1>%SegmentIteratorPrototype%.following( [ _from_ ] )</h1>
+        <p>The `following` method moves the iterator index to the boundary following the code unit index _from_ (or after the current index of the iterator if _from_ is *undefined*) and returns a Boolean value that is *true* if and only if the end of the string was reached. It performs the following steps:</p>
```

Comment:
I guessed the intended semantics of `following` and `preceding` return values from README text. They appear to be based on `done` values in iterator results returned by `next()`, _except_ that they return true when reaching an endpoint of the string **after** finding a boundary rather than returning true when already at the end **before** seeking one. To be honest, this seems like a bit of a [boolean trap](https://ariya.io/2011/08/hall-of-api-shame-boolean-trap), and I'm not sure what patterns it is expected to support anyway.

For example, we could just as easily rely on the RangeError thrown for out-of-bounds iteration, or introduce something like `%IteratorPrototype%.atEndpoint()`, or even just invert the return value to indicate "more content available" rather than "terminal boundary reached", which may read slightly more naturally in English because "preceding" and "following" have boolean connotations (e.g. `moreFollows = boundaries.following()` vs. `done = boundaries.following()`):
```js
// findWord searches a string for a specified whole word from a specified index,
// wrapping around if necessary.
// It returns the index at which the found word begins, or -1 if it is not found.
function findWord( string, word, start ) {
  let boundaries = segmenter.segment(string);

  // Begin at the first boundary at or after `start`.
  boundaries.following(start);
  boundaries.preceding();
  let moreFollows = true;
  if ( boundaries.index < start ) {
    𝗺𝗼𝗿𝗲𝗙𝗼𝗹𝗹𝗼𝘄𝘀 = 𝗯𝗼𝘂𝗻𝗱𝗮𝗿𝗶𝗲𝘀.𝗳𝗼𝗹𝗹𝗼𝘄𝗶𝗻𝗴();
  }
  let startIndex = boundaries.index;
  let lastIndex = startIndex;
  
  while ( moreFollows ) {
    𝗺𝗼𝗿𝗲𝗙𝗼𝗹𝗹𝗼𝘄𝘀 = 𝗯𝗼𝘂𝗻𝗱𝗮𝗿𝗶𝗲𝘀.𝗳𝗼𝗹𝗹𝗼𝘄𝗶𝗻𝗴();
    let nextIndex = boundaries.index;
    if ( string.slice(lastIndex, nextIndex) === word ) {
      return lastIndex;
    }
    lastIndex = nextIndex;
  }
  
  // Wrap around from the end of string until re-reaching `startIndex`.
  boundaries.preceding(1);
  lastIndex = 0;
  moreFollows = true;
  while ( moreFollows ) {
    boundaries.following();
    let nextIndex = boundaries.index;
    moreFollows = nextIndex < startIndex;
    if ( string.slice(lastIndex, nextIndex) === word ) {
      return lastIndex;
    }
    lastIndex = nextIndex;
  }
  
  return -1;
}
```

Reply:
Should we pursue this idea in a separate issue/discussion at the ECMA-402 meeting? I don't quite understand the motivation for this change. But you interpreted the preceding/following methods how I intended them.

Reply:
I guess we could talk about it briefly. There's currently also an interesting edge case with empty strings, for which any index argument to `preceding` or `following` results in a RangeError.